test.py

```diff
@@ -1313,7 +1313,12 @@ def write_consolidated_boost_junit_xml(tmpdir: str, mode: str) -> None:
         for test in suite.tests:
             if test.mode != mode:
                 continue
-            test_xml = test.get_junit_etree()
+            try:
+                test_xml = test.get_junit_etree()
+            except ET.ParseError as e:
+                print(
+                    palette.warn(f'Failed to parse XML for {test.uname} with error: {e}')
+                )
```

Comment:
The patch is looking good, but don't you think the same error should make its way into the consolidated XML, we should not just quietly skip the test?

Reply:
Ah, maybe we have an empty file if the test crashed. It should certainly be recorded as a crash.

Reply:
> The patch is looking good, but don't you think the same error should make its way into the consolidated XML, we should not just quietly skip the test?

@kostja not sure I got your question, this print is on the consolidation function which runs after all tests are done, it only prints a WARN to notify the user but continues parsing other XMLs

Reply:
We should create an xml fragment in the consolidated xml that indicates a failure. So it's visible in the test result analyzer.

Reply:
Beni, see Avi's explanation.

Reply:
@avikivity there is a long standing problem that xml output can get corrupt if boost logging to xml is used from multiple threads. This is another reason of a failure to parse.